.buildkite/packaging.pipeline.yml

```diff
@@ -0,0 +1,229 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+# TODO: Pre-cache beats-dev/golang-crossbuild container image
+
+env:
+  ASDF_MAGE_VERSION: 1.15.0
+  AWS_ARM_INSTANCE_TYPE: "m6g.xlarge"
+  AWS_IMAGE_UBUNTU_ARM_64: "platform-ingest-beats-ubuntu-2204-aarch64"
+  GCP_DEFAULT_MACHINE_TYPE: "c2d-highcpu-8"
+  IMAGE_UBUNTU_X86_64: "family/platform-ingest-beats-ubuntu-2204"
+
+steps:
+  - group: Beats dashboards
+    key: dashboards
+    steps:
+      - label: Snapshot dashboards
+        key: dashboards-snapshot
+        # TODO: container with go and make
+        agents:
+          provider: gcp
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        commands:
+          - make build/distributions/dependencies.csv
+          - make beats-dashboards
+        env:
+          SNAPSHOT: true
+        artifact_paths:
+          - build/distributions/**/*
+
+      - label: Staging dashboards
+        if: "build.branch =~ /\\d+\\.\\d+/ || build.branch == 'main'"
+        key: dashboards-staging
+        # TODO: container with go and make
+        agents:
+          provider: gcp
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        commands:
+          - make build/distributions/dependencies.csv
+          - make beats-dashboards
+        artifact_paths:
+          - build/distributions/**/*
+
+  - group: Packaging snapshot
+    key: packaging-snapshot
+    steps:
+      # x86
+      - label: ":ubuntu: {{matrix}}/Packaging Linux Snapshot"
+        env:
+          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
```

Comment:
Do you think a top-level env variable with the different combinations could help?

```
env:
  PLATFORMS_AMD: " "
  PLATFORMS_ARM: " "

...

- label: "..."
  env:
    PLATFORMS: "${PLATFORMS_AMD}"
...
```

Reply:
Btw this is probably to the approach taken for migrated pipelines to be explicit when the variable is referenced in just a few steps e.g. https://github.com/elastic/beats/blob/main/.buildkite/x-pack/pipeline.xpack.heartbeat.yml#L231

If, we found ourselves in this pipeline that we need to copy/paste it in most places, agreed, it makes sense to reduce the copy/pasting.

Reply:
Extracted the variables